smdebug/tensorflow/keras.py

```diff
@@ -620,3 +623,129 @@ def new_add_weight(opt, *args, **kwargs):
         # Optimizer is being saved to support additional features in the future.
         self.optimizer = optimizer
         return optimizer
+
+    def _unwrap_tape(self):
+        def _is_wrapper(f):
+            return hasattr(f, "__wrapped__")
+
+        def unwrap(func):
+            while _is_wrapper(func):
+                func = func.__wrapped__
+            return func
+
+        self.tape.__class__._push_tape = unwrap(self.tape.__class__._push_tape)
+        self.tape.__class__._pop_tape = unwrap(self.tape.__class__._pop_tape)
+        self.tape.__class__.gradient = unwrap(self.tape.__class__.gradient)
+
+    def close(self):
```

Comment:
@jarednielsen Adding this in close() will require the user to call close() for sure.
Do you think unwrap_tape is more suitable in _cleanup(self) ?

Reply:
Yes

Reply:
done in the latest commit